File: ios/MyHelloModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MyHelloModule, NSObject)
RCT_EXTERN_METHOD(helloWorld:(NSString *)name
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/MyHelloModule.swift
import Foundation

@objc(MyHelloModule)
class MyHelloModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func helloWorld(_ name: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    do {
      let result = "Hello, \(name) from Native!"
      resolver(result)
    } catch {
      rejecter("ERROR", error.localizedDescription, error)
    }
  }
}